libs/plugins/stream-client/src/lib/core/state/graph-to-tiles.ts

```diff
@@ -0,0 +1,32 @@
+import { DiagramGraph } from '@flogo-web/lib-client/core';
+import { TaskTile, TileType } from '@flogo-web/lib-client/diagram';
+
+export const graphToTiles = (graph: DiagramGraph, maxTileCount: number) => {
+  let currentId = graph.rootId;
+  const tiles: TaskTile[] = [];
+  while (currentId) {
+    const currentStage = graph.nodes[currentId];
+    if (
+      tiles.length >= maxTileCount - 1 &&
+      currentStage &&
+      !currentStage.children.length
```

Comment:
What if there are more stages / children but the tiles count is fixed? I think they will still be in the pipeline but not available for configuration.